### DISCUSSION TITLE
Error TLS: failed to find any PEM data in certificate input

### DISCUSSION THREAD

**Discussion by adeifeadekunbi** [hoophq/hoop#619] [Q&A] [answered]
Hi Team,

I am attempting to deploy Hoop.dev using Docker Compose with TLS enabled. I have configured the environment variables as follows:

- TLS_KEY=/etc/cass/tls/key.pem
- TLS_CERT=/etc/cass/tls/cert.pem

However, when I execute docker-compose up, I encounter the following error:
`gateway-1  | 2024-12-27T09:54:34Z       fatal   gateway/main.go:54      failed parsing key pair, err=tls: failed to find any PEM data in certificate input
gateway-1  | github.com/hoophq/hoop/gateway.Run
gateway-1  |    /home/runner/work/hoop/hoop/gateway/main.go:54
`
To debug, I tested the loadServerCertificates() function locally on my host system, and it successfully loaded the certificates. This confirms that the certificates are valid.

A copy of my docker-compose.yml file is attached for reference. Could you help review the configuration and suggest potential solutions to resolve this issue?
[dockercompose hoop.txt](https://github.com/user-attachments/files/18259711/dockercompose.hoop.txt)

**Comment by sandromello** [comment, marked as answer]
Hello there,

You need to use the syntax `file://` or `base64://` to load it properly:

- `file:///etc/cass/tls/key.pem`
- `base64://<base64-content>`

**Comment by adeifeadekunbi** [comment]
Thank you it worked
